From cd32c9d1ba863f5cff280c4b1fd31da516869242 Mon Sep 17 00:00:00 2001
From: Sebastian Walderich <sebastian@walderich.de>
Date: Mon, 23 Jul 2012 10:33:11 +0200
Subject: [PATCH] Disable completion of files without DynamicLibrary

Under Windows the auto completion of file names is done via wxDynamicLibrary
class. If this is not enabled, the corresponding method is now disabled as
well.
---
 include/wx/msw/textentry.h |    2 ++
 src/msw/textentry.cpp      |    4 ++++
 2 files changed, 6 insertions(+)

diff --git a/include/wx/msw/textentry.h b/include/wx/msw/textentry.h
index c42ef8d..511797d 100644
--- a/include/wx/msw/textentry.h
+++ b/include/wx/msw/textentry.h
@@ -75,7 +75,9 @@ protected:
     // wxUSE_OLE as OleInitialize() is not called then
 #if wxUSE_OLE
     virtual bool DoAutoCompleteStrings(const wxArrayString& choices);
+#if wxUSE_DYNLIB_CLASS
     virtual bool DoAutoCompleteFileNames(int flags);
+#endif // wxUSE_DYNLIB_CLASS
     virtual bool DoAutoCompleteCustom(wxTextCompleter *completer);
 #endif // wxUSE_OLE
 
diff --git a/src/msw/textentry.cpp b/src/msw/textentry.cpp
index 02936ea..67e03be 100644
--- a/src/msw/textentry.cpp
+++ b/src/msw/textentry.cpp
@@ -744,6 +744,8 @@ void wxTextEntry::GetSelection(long *from, long *to) const
 
 #ifdef HAS_AUTOCOMPLETE
 
+#if wxUSE_DYNLIB_CLASS
+
 bool wxTextEntry::DoAutoCompleteFileNames(int flags)
 {
     typedef HRESULT (WINAPI *SHAutoComplete_t)(HWND, DWORD);
@@ -791,6 +793,8 @@ bool wxTextEntry::DoAutoCompleteFileNames(int flags)
     return true;
 }
 
+#endif // wxUSE_DYNLIB_CLASS
+
 wxTextAutoCompleteData *wxTextEntry::GetOrCreateCompleter()
 {
     if ( !m_autoCompleteData )
-- 
1.7.10.msysgit.1

